packages/wonder-blocks-icon/components/icon.js

```diff
@@ -0,0 +1,130 @@
+     * or `xlarge` (96px).
+     */
+    size: IconSize,
+    /**
+     * Styles that can be processed by `addStyle` — bare style objects,
+     * Aphrodite style objects, or arrays thereof.
+     */
+    style?: any,
+};
+
+const StyledSVG = addStyle("svg");
+
+const viewportPixelsForSize: {[IconSize]: number} = {
+    small: 16,
+    medium: 24,
+    large: 48,
+    xlarge: 96,
+};
+
+const getPathForIcon = (
+    name: IconName,
+    size: IconSize,
+): {
+    assetSize: IconSize,
+    path: string,
+} => {
+    if (!iconPaths[name]) {
+        throw new Error(`There is no icon named ${name}.`);
+    }
+    const paths = iconPaths[name];
+    if (paths && paths[size]) {
+        // Great, we have the IconSize we actually requested
+        return {assetSize: size, path: paths[size]};
+    } else {
+        // Oh, no, we don't have the right IconSize! Let's find the next best
+        // one...we prefer
+        const desiredPixelSize = viewportPixelsForSize[size];
+        const availableSizes = Object.keys(paths);
+        const assetSizes = availableSizes.sort((availableSize) => {
+            const availablePixelSize = viewportPixelsForSize[availableSize];
+            const sizeDifference = availablePixelSize - desiredPixelSize;
+            return sizeDifference < 0 ? sizeDifference : 1;
+        });
+        const bestAssetSize = assetSizes[0];
+        if (bestAssetSize && paths[bestAssetSize]) {
+            return {assetSize: bestAssetSize, path: paths[bestAssetSize]};
+        } else {
+            throw new Error(
+                `No path available for icon ${name} at size ${size}`,
+            );
+        }
+    }
+};
+
+/**
+ * An Icon displays a small informational or decorative image as an SVG. Wonder
+ * Blocks has a fixed set of icons and icon sizes.
+ *
+ * ```js
+ * <Icon
+ *     name={correct}
+ *     color={Colors.green}
+ *     size={"medium"}
+ *     style={{margin: 4}}
+ * />
+ * ```
+ * @version 0.0.1
+ */
+export default class Icon extends React.PureComponent<Props> {
+    static defaultProps = {
+        color: Color.offBlack,
+    };
+
+    render() {
+        const {color, size, name, style} = this.props;
+        const {assetSize, path} = getPathForIcon(name, size);
+        const pixelSize = viewportPixelsForSize[size];
+        const viewboxPixelSize = viewportPixelsForSize[assetSize];
+        return (
+            <StyledSVG
+                style={[styles.svg, style]}
+                width={pixelSize}
+                height={pixelSize}
+                // There is a weird thing where Flow will only recognize a
+                // string-quoted prop name if it's in single quotes, but our
+                // tooling normalizes it to double-quotes on commit. So the
+                // aria-label prop isn't included in props validation.
+                // eslint-disable-next-line react/prop-types
+                aria-label={this.props["aria-label"]}
```

Comment:
Can you open an issue with the eslint react plugin and link to the issue here?  It seems like this should be fixable since it's a string literal.

Reply:
I can look into it, but I would need to spend some time writing up a minimal repro. I think that's outside the scope of what I need to get done this sprint.

Reply:
This captured in https://khanacademy.atlassian.net/browse/WB-291.